spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -848,6 +849,16 @@ public void testRewriteWithUntranslatedOrUnconvertedFilter() {
         .hasMessageContaining("Cannot convert Spark filter");
   }
 
+  @Test
+  public void testAddApplicationIdInSummary() {
+    createPartitionTable();
+    // create 5 files for each partition (c2 = 'foo' and c2 = 'bar')
+    insertData(10);
+    sql("CALL %s.system.rewrite_data_files(table => '%s')", catalogName, tableIdent);
+
+    assert (snapshotSummary().containsKey(CatalogProperties.APP_ID));
```

Comment:
nit: we can probably add this validation in `testRewriteDataFilesOnPartitionTable` itself. We may not want to increase the CI time by creating table, loading data and rewriting just to validate this.